pkg/telemetry/prom_cass_servicemonitor.go

```diff
@@ -13,17 +13,22 @@ import (
 )
 
 const (
-	CassandraMetricsPort = 9103
+	// CassandraMetricsPortLegacy is the metrics port to scrape for the legacy MCAC stack (Metrics
+	// Collector for Apache Cassandra).
+	CassandraMetricsPortLegacy = 9103
+	// CassandraMetricsPortModern is the metrics port to scrape for the modern stack (metrics
+	// exposed by management-api).
+	CassandraMetricsPortModern = 9000
 )
 
-// Static configuration for ServiceMonitor's endpoints.
-const endpointString = `
+// Static configuration for ServiceMonitor's endpoints when using the legacy MCAC stack (Metrics
+// Collector for Apache Cassandra).
+const endpointStringLegacy = `
 apiVersion: monitoring.coreos.com/v1
 kind: ServiceMonitor
 spec:
   endpoints:
-  - port: prometheus
-    interval: 15s
+  - interval: 15s
```

Comment:
There was something fishy here. We were defining both `port` and `targetPort` but the specs say we should define one or the other, not both. Besides, `prometheus` port is 9090, which is not what we want in any case (MCAC enabled or disabled). So I went ahead and removed `port: prometheus`. Unfortunately there is no e2e or kuttl test to verify this, so I'm going to do some manual testing around this today.